web/src/models/KafkaRequest.model.ts

```diff
@@ -19,6 +19,10 @@ const KafkaRequest = ({
   messageKey = '',
   messageValue = '',
 }: TRawKafkaRequest): KafkaRequest => {
+  if (!headers) {
```

Comment:
We try to avoid side effects like this, I think it is better to have something like
```
headers: !headers ? [] : ...
```

Reply:
It makes sense. I've updated this code to something similar, using the notation `(headers || [])` when building the model: 
```js

    headers: (headers || []).map(({key = '', value = ''}) => ({
```